lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -181,12 +186,266 @@ def test_duplicate(self, selinux_disabled, server_config, make_logger, tarball):
+            else:
+                assert args[0] == "find"
+                raise subprocess.TimeoutExpired("tar", 43)
+
+        monkeypatch.setattr("subprocess.run", mock_run)
+        tree = FileTree(server_config, make_logger)
+        tree.create("ABC", source_tarball)
+
+        tarball_name = source_tarball.name.removesuffix(".tar.xz")
+        unpack_dir = server_config.INCOMING / "ABC" / f"{tarball_name}.unpack"
+
+        tarball = tree.find_dataset(md5)
+        with pytest.raises(TarballModeChangeError) as exc:
+            tree.unpack(md5)
+        assert (
+            str(exc.value)
+            == f"An error occurred while changing file permissions of {unpack_dir}: Command 'tar' timed out after 43 seconds"
+        )
+
+    def test_unpack_find_error(
+        self, monkeypatch, selinux_enabled, server_config, make_logger, tarball
+    ):
+        """Show that, when returncode of the Tarball permission change subprocess is not
+        zero, it is handled successfully."""
+        source_tarball, source_md5, md5 = tarball
+
+        def mock_run(args, **kwargs):
+            if args[0] == "tar":
+                return subprocess.CompletedProcess(
+                    args, returncode=0, stdout="Successfully Unpacked!", stderr=None
+                )
+            else:
+                assert args[0] == "find"
+                return subprocess.CompletedProcess(
+                    args,
+                    returncode=1,
+                    stdout=None,
+                    stderr="Some error unpacking tarball\n",
+                )
+
+        monkeypatch.setattr("subprocess.run", mock_run)
+        tree = FileTree(server_config, make_logger)
+        tree.create("ABC", source_tarball)
+
+        tarball = tree.find_dataset(md5)
+        tarball_name = source_tarball.name.removesuffix(".tar.xz")
+        unpack_dir = server_config.INCOMING / "ABC" / f"{tarball_name}.unpack"
+
+        with pytest.raises(TarballModeChangeError) as exc:
+            tree.unpack(md5)
+        assert (
+            str(exc.value)
+            == f"An error occurred while changing file permissions of {unpack_dir}: tar exited with status 1:  'Some error unpacking tarball'"
+        )
+
+    def test_unpack_move_src(
+        self, monkeypatch, selinux_enabled, server_config, make_logger, tarball, caplog
+    ):
+        """Show that, when unpack directory for moving results is not present and
+        raises FileNotFoundError Exception, it is handled successfully."""
+        source_tarball, source_md5, md5 = tarball
+
+        def mock_move(src: pathlib.Path, dest: pathlib.Path):
+            raise FileNotFoundError(f"No such file or directory: '{src}'")
+
+        def mock_run(args, **kwargs):
+            return subprocess.CompletedProcess(
+                args, returncode=0, stdout="Success!", stderr=None
+            )
+
+        monkeypatch.setattr("subprocess.run", mock_run)
+        monkeypatch.setattr(shutil, "move", mock_move)
```

Comment:
When the `shutil.move()` fails, the CUT is going to call `shutil.rmtree()`, so you should have a mock for that, here.  Ditto at line 389.